features/search-replace.feature

```diff
@@ -1106,6 +1106,28 @@ Feature: Do global search/replace
       a:1:{i:0;O:10:"CornFlakes":0:{}}
       """
 
+  # Regression test for https://github.com/wp-cli/search-replace-command/issues/191
+  Scenario: Deserialization for empty, type-hinted objects are handled gracefully
+
+    Given a WP install
+    And I run `wp option add cereal_isation 'O:13:"mysqli_result":5:{s:13:"current_field";N;s:11:"field_count";N;s:7:"lengths";N;s:8:"num_rows";N;s:4:"type";N;}'`
```

Comment:
It seems a little brittle to add a serialized object to the database as a string. I guess we can't instantiate the object in PHP and save it though, because we'll run into the same type error.

Can we use `wp db query` directly so we bypass WordPress APIs and can make sure we're making a direct entry into the database?

Also, we should add a second serialized object that deserializes correctly, and make sure it's replaced appropriately.

Reply:
exactly, can't really avoid it if all object injection via deserialization is strictly typed now in 8.1.

Gotcha, can do both of those test changes

Reply:
@danielbachhuber been running into an problem w/ trying to make a second serialized object that would work similar to this case. If you have a `mysql_result` that is not empty IE:

```
object(mysqli_result)#2 (5) {
  ["current_field"]=>
  int(0)
  ["field_count"]=>
  int(1)
  ["lengths"]=>
  NULL
  ["num_rows"]=>
  int(0)
  ["type"]=>
  int(1)
}
```

if you try to serialize that in PHP 8.3 you end up with this:

```
string(25) "O:13:"mysqli_result":0:{}"
```

which makes sense - they've forced `readonly` properties on classes to not serialize, but it makes an empty object that PHP considers not empty, not null, iterable & not a `\__PHP_Incomplete_Class`.

The object itself is truly empty which confuses the `foreach` here: 
https://github.com/wp-cli/search-replace-command/blob/2f48d9fb2fda523524740fc91eb402b496aa941b/src/WP_CLI/SearchReplacer.php#L110-L112

what you get in response is:

```
PHP Fatal error:  Uncaught Error: mysqli_result object is already closed in src/WP_CLI/SearchReplacer.php:123
```

the only remedy I've cooked up is casting `$data` to an array in the loop so far:

`foreach ( ( array ) $data as $key => $value ) {` 

all tests pass in behat with this change. Curious if you have other thoughts and ideas I could try.